src/mappings/handlers/multisigEventHandler.ts

```diff
@@ -0,0 +1,157 @@
+import { SubstrateEvent } from "@subql/types";
+import { u8aToHex } from "@polkadot/util";
+import { decodeAddress } from "../../utils";
+import { CreateCallVisitorBuilder, CreateCallWalk, VisitedCall } from "subquery-call-visitor";
+import { EventStatus, MultisigEvent, MultisigOperation, OperationStatus } from "../../types";
+import {
+  generateEventId,
+  generateOperationId,
+  getBlockCreated,
+  getDataFromCall,
+  getDataFromEvent,
+  getIndexCreated,
+  timestamp,
+} from "../../utils/operations";
+import { AccountId, DispatchResult, Timepoint } from "@polkadot/types/interfaces";
+import { AnyTuple, CallBase } from "@polkadot/types/types";
+
+const callWalk = CreateCallWalk();
+
+export async function handleMultisigCallEvent(event: SubstrateEvent) {
+  if (!event.extrinsic) throw new Error("Extrinsic not found");
+
+  const isNewMultisig = event.event.method === "NewMultisig";
+
+  const possibleIndex = isNewMultisig ? 2 : 3;
+  const callHash = getDataFromEvent<Uint8Array>(event.event, "callHash", possibleIndex);
+
+  if (!callHash) throw new Error("Call hash not found");
+
+  const callHashString = u8aToHex(callHash);
+
+  const multisigPossibleIndex = isNewMultisig ? 1 : 2;
+  const multisig = getDataFromEvent<AccountId>(event.event, "multisig", multisigPossibleIndex);
+
+  if (!multisig) throw new Error("Multisig not found");
+
+  const multisigAccountId = multisig.toHex();
+
+  let timepoint = null;
+  let blockCreated = getBlockCreated(event.extrinsic);
+  let indexCreated = getIndexCreated(event.extrinsic);
+  if (!isNewMultisig) {
+    timepoint = getDataFromEvent<Timepoint>(event.event, "timepoint", 1) || getDataFromEvent<Timepoint>(event.event, "Timepoint", 1);
+  }
+  if (timepoint && timepoint.height && timepoint.index) {
+    blockCreated = timepoint.height.toNumber();
+    indexCreated = timepoint.index.toNumber();
+  }
+
+  const existingOperation = await findExistingOperation(callHashString, blockCreated, indexCreated, multisigAccountId);
+
+  const operationId = generateOperationId(callHashString, multisigAccountId, blockCreated, indexCreated);
+
+  const newOperation = await MultisigOperation.create({
+    ...existingOperation,
+    id: operationId,
+    callHash: callHashString,
+    status: existingOperation?.status || OperationStatus.pending,
+    accountId: multisigAccountId,
+    depositor: u8aToHex(decodeAddress(event.extrinsic.extrinsic.signer.toString())),
+    blockCreated: blockCreated,
+    indexCreated: indexCreated,
+    timestamp: timestamp(event.extrinsic.block),
+  });
+
+  await newOperation.save();
+
+  const handleCall = async (visitedCall: VisitedCall) => {
+    const call = getDataFromCall<CallBase<AnyTuple>>(visitedCall.call, "call");
+
+    const existingOperation = await findExistingOperation(callHashString, blockCreated, indexCreated, multisigAccountId);
```

Comment:
We cannot really guarantee that the call we are processing correponds to the same callhash. Imagine the following situation:

batch(as_multi(msig1, call1), as_multi(msig2, call2))

Such a call will emit two NewMultisig events and when we will process the first one, we might wrongly set its call-data to be equal to the call-data of the second call. I think we should calculate the hash of the `call` and only set the call-data in case the calculated call-hash matches call-hash from the event we are currently processing. This way, we will only modify call-data for each even once